_layouts/tutorial.html

```diff
@@ -18,120 +18,126 @@
+          {%- assign file_name = file_name[1] -%}
+
+          {%- assign p_file_name = p.path | split: "/" -%}
+          {%- assign p_file_name = p_file_name[1] -%}
+
+          {%- assign topic = file_name | split: "-" | first -%}
+          {%- assign p_topic = p_file_name | split: "-" | first -%}
+
+          {%- if p.lang == page.lang -%}
+            {%- assign step_index = step_index | plus: 1 -%}
+            {%- if p_topic == topic and same_topic_started == false -%}
               <div class="same-topic">
-              {% assign same_topic_started = true %}
-            {% endif %}
-            {% if p_topic != topic and same_topic_started == true %}
+                {% comment %}{% endcomment %}
+              {%- assign same_topic_started = true -%}
+            {%- endif -%}
+            {%- if p_topic != topic and same_topic_started == true -%}
               </div>
-              {% assign same_topic_started = false %}
-            {% endif %}
-            <a id="page-{{ p.path }}" class="step {% if page == p %}current{% endif %}" href="..{{ p.url }}">{% if page == p %}{{ step_index }}{% endif %}</a>
-            {% if should_assign_next %}
-              {% assign should_assign_next = false %}
-              {% assign has_next_page = true %}
-              {% assign next_page = p %}
-            {% endif %}
-            {% if page == p %}
-              {% assign should_assign_next = true %}
-            {% endif %}
-          {% endif %}
-        {% endfor %}
-        {% if same_topic_started %}
+              {% comment %}{% endcomment %}
+              {%- assign same_topic_started = false -%}
+            {%- endif -%}
+            <a id="page-{{- p.path -}}" class="step {%- if page == p -%}current{%- endif -%}" href="..{{- p.url -}}">{%- if page == p -%}{{- step_index -}}{%- endif -%}</a>
+            {% comment %}{% endcomment %}
+            {%- if should_assign_next -%}
+              {%- assign should_assign_next = false -%}
+              {%- assign has_next_page = true -%}
+              {%- assign next_page = p -%}
+            {%- endif -%}
+            {%- if page == p -%}
+              {%- assign should_assign_next = true -%}
+            {%- endif -%}
+          {%- endif -%}
+        {%- endfor -%}
+        {%- if same_topic_started -%}
           </div>
-        {% endif %}
+          {% comment %}{% endcomment %}
+        {%- endif -%}
       </div>
-      <h2>{{ page.title | escape }}</h2>
+      {% comment %}{% endcomment %}
+      <h2>{{- page.title | escape -}}</h2>
     </header>
     <div class="main">
-      {{ content }}
-      {% if has_next_page %}
-        <a href="..{{ next_page.url }}" class="next-page">{{ site.data.localization.footer.nextStep[page.lang] }}</a>
-      {% endif %}
+      {{- content -}}
+      {%- if has_next_page -%}
+        <a href="..{{- next_page.url -}}" class="next-page">{{- site.data.localization.footer.nextStep[page.lang] -}}</a>
```

Comment:
Adding the `-` has no effect here. Did you do it for consistency?

Reply:
This reply will be a reply to all the threads here.  When editing the layout file, I did two things:

1. **I search-replaced the following strings:**
    * ```"{{ " --> "{{- "```
    * ```"{% " --> "{%- "```
    * ```" }}" --> " -}}"```
    * ```" %}" --> " -%}"```

This took out all the whitespace between elements. However, it does not save indentation, as you requested in the [solution](https://github.com/CoderDojoPotsdam/regex-tutorial/issues/111):
>Please use this to reduce the number spaces.
>When you do that, please keep the indentation intact - each element should be on a new line.

Taking out all the whitespace put each ```<a>``` in the ```<header>``` on the same line.  I had to add some space back so that each ```<a>``` had its own line.

2. **I added comments to make spaces after certain html elements inside the ```<header>```.**

Before these changes, the page source looked like this: view-source:https://coderdojopotsdam.github.io/regex-tutorial/en/00-01.html?

After the changes, the page source looked like this: view-source:https://undoingtech.github.io/regex-tutorial/en/00-01.html?

Note: I cannot provide a link directly to the source code, so you will have to view it yourself after you click on the link.